smarts/sstudio/genhistories.py

```diff
@@ -279,19 +282,33 @@ def column_val_in_row(self, row, col_name: str):
                 dm = np.linalg.norm((dx, dy))
                 if dm > 0.0:
                     r = math.atan2(dy / dm, dx / dm)
-                    return (r - math.pi / 2) % (2 * math.pi)
+                    new_heading = (r - math.pi / 2) % (2 * math.pi)
+                    if self._max_angular_velocity:
+                        # XXX: could try to divide by sim_time delta here instead of assuming .1s
+                        angular_velocity = (new_heading - self._prev_heading) / 0.1
+                        if abs(angular_velocity) > self._max_angular_velocity:
+                            new_heading = (
+                                self._prev_heading
+                                + np.sign(angular_velocity)
+                                * self._max_angular_velocity
+                                * 0.1
+                            )
```

Comment:
`if abs(angular_velocity) > self._max_angular_velocity`, shouldn't the `self._max_angular_velocity` be updated to the new `angular_velocity` ?

Reply:
No, b/c this is a parameter set by the user that should remain fixed.  I.e., we're not trying to _find_ the maximum here, rather, this is the maximum _allowable_.  This is a way of filtering out "noise" in the dataset positions.

Reply:
I see. Makes sense!